File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/ProofOfPaymentDto.cs
using ProjectPos.Data.Shared.Enums;
using ProjectPos.Services.EntityDtos;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace ProjectPos.Services.DTOs
{
    public class ProofOfPaymentDto : FullAuditedEntityDto<int>
    {
        public string? Reference { get; set; }
        public int? CustomerId { get; set; }
        public string? Bank { get; set; }
        public string? CustomerName { get; set; }
        public string? CreatorName { get; set; }
        public string? BranchCode { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public Currency? Currency { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? PaidAmount { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? UsableAmount { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? UsedAmount { get; set; }
        public DateTime? BankingDate { get; set; }
    }
}

File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/proof-of-payment-dto.ts
import { FullAuditedEntityDto } from "../entity-dtos/full-audited-entity-dto";
import { Currency } from "../enums/currency";
import { CompanyDto } from "./company-dto";
import { SalesOrderDto } from "./sales-order-dto";

export interface ProofOfPaymentDto extends FullAuditedEntityDto<number> {
    reference?: string;
    customerId?: number;
    bank?: string;
    customerName?: string;
    creatorName?: string;
    branchCode?: string;
    currency?: Currency;
    paidAmount?: number;
    usableAmount?: number;
    usedAmount?: number;
    bankingDate?: Date;
    customer?: CompanyDto
}
